### ISSUE TITLE
Weird bug during upgrade of 10y-old GLPI instance

### ISSUE THREAD

**Issue by epgfm** [glpi-project/glpi#12053]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.1

### Bug description

During upgrade of an extremely old instance, the upgrade process of the database stalled at that point:
![image](https://user-images.githubusercontent.com/10635269/176491321-8e61857d-aa99-4c76-a178-83561be7b59f.png)

I looked at the source to see where that error came from and tracked it down to the buildDelete method used by the deleteOrDie method from [DBmysql](https://github.com/glpi-project/glpi/blob/10.0/bugfixes/src/DBmysql.php).
Now because my eyes aren't used to SQL delete queries and the docstring of the method is wrong (parameter list), this line looked weird:
```php
$query  = "DELETE " . self::quoteName($table) . " FROM " . self::quoteName($table);
```
So I changed it to:
```php
$query  = "DELETE FROM " . self::quoteName($table);
```
Basically undoing commit 4685a53.
After doing that, the update ran with no issue to completion.
Here's the thing tho, I don't understand why.
After looking into it, it turns out
```sql
DELETE `table` from `table` where blabla
```
is actually valid SQL syntax, so my change shouldn't have made a difference.
I'm stumped, so I thought I'd make a bug report.
I don't have all the system info requested in the issue template so I'll gather that tomorrow.

**Comment by cedric-anne** [comment]
Hi,

This error is not related to GLPI itself. It is an error that seems to indicate that your database files are corrupted. You should probably try to restore your dump and check that everything is OK, then process to GLPI update.

See https://www.stellarinfo.com/blog/mysql-error-incorrect-key-file-on-table/ for some technical explanations about this MySQL issue.

I close this issue as we cannot do anything for it on GLPI side.
